### ISSUE TITLE
AWS full terraform install ws-proxy image not found

### ISSUE THREAD

**Issue by daimor** [gitpod-io/gitpod#2141]
I can't install gitpod to AWS, got this error for ws-proxy

 Failed to pull image "gcr.io/gitpod-io/ws-proxy:v0.4.0": rpc error: code = Unknown desc = Error response from daemon: manifest for gcr.io/gitpod-io/ws-proxy:v0.4.0 not found: manifest unknown: Failed to fetch "v0.4.0" from request "/v2/gitpod-io/ws-proxy/manifests/v0.4.0".

```
dashboard-57b7c5f766-ts7rl          1/1     Running            0          27m
gitpod-db-initialization-d87xm      0/1     Completed          0          31m
image-builder-576b8b6488-ws9rd      2/2     Running            0          27m
messagebus-65bddbf7b4-4t2v6         1/1     Running            0          27m
minio-87788f64c-8fp9x               1/1     Running            0          31m
node-daemon-6hfm6                   1/1     Running            0          27m
node-daemon-k8wdk                   1/1     Running            0          27m
node-daemon-mmtlr                   1/1     Running            0          27m
node-daemon-mzvj8                   1/1     Running            0          27m
node-daemon-pfcml                   1/1     Running            0          27m
node-daemon-tm7t6                   1/1     Running            0          27m
proxy-689b5ddf96-fnl4l              0/1     CrashLoopBackOff   9          27m
registry-facade-56cf59b97d-klm9p    0/1     CrashLoopBackOff   10         27m
server-64cd6b6dfd-clzzx             1/1     Running            0          27m
theia-server-77ddd4d96c-ln7vf       1/1     Running            0          27m
ws-daemon-6ffwb                     0/1     Init:0/1           0          27m
ws-daemon-d7rn2                     0/1     Init:0/1           0          27m
ws-daemon-dqtgj                     0/1     Init:0/1           0          27m
ws-daemon-fmjp7                     0/1     Init:0/1           0          27m
ws-daemon-ptr8f                     0/1     Init:0/1           0          27m
ws-daemon-q2wxt                     0/1     Init:0/1           0          27m
ws-manager-6bf5684759-lx6rh         1/1     Running            0          27m
ws-manager-bridge-974fb844c-wb6js   1/1     Running            0          27m
ws-proxy-774b7c56f6-srh9l           0/1     ImagePullBackOff   0          27m
ws-scheduler-647884bf56-6fqbp       1/1     Running            0          27m
```

**Comment by hm2075** [comment]
Is this the same  https://github.com/gitpod-io/gitpod/issues/2125 ??

**Comment by daimor** [comment]
Not sure, about it. I'm just using this instruction
https://github.com/gitpod-io/gitpod/blob/master/install/aws-full-terraform/README.md

In my case, I just get an error, with no image `gcr.io/gitpod-io/ws-proxy:v0.4.0`
Would it be possible to check and push the missing image?

**Comment by daimor** [comment]
Well, I've updated values.yml, and set version: 0.5.0, now I've successfully pulled image
But, ws-proxy now crashes with error
```
{"message":"enabled JSON logging","serviceContext":{"service":"ws-proxy","version":""},"severity":"info","time":"2020-11-08T14:35:35Z"}
{"message":"enabled verbose logging","serviceContext":{"service":"ws-proxy","version":""},"severity":"info","time":"2020-11-08T14:35:35Z"}
{"@type":"type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent","error":"config validation error:\n    github.com/gitpod-io/gitpod/ws-proxy/cmd.getConfig\n        /tmp/build/components-ws-proxy--app.ca3b7dcc9d8c3c1634ff4e7f1297c63007ce8d00/cmd/root.go:203\n  - TheiaServer not configured:\n    github.com/gitpod-io/gitpod/ws-proxy/pkg/proxy.(*TheiaServer).Validate\n        /tmp/build/components-ws-proxy--app.ca3b7dcc9d8c3c1634ff4e7f1297c63007ce8d00/pkg/proxy/config.go:102","filename":"/config/config.json","message":"cannot load config","serviceContext":{"service":"ws-proxy","version":""},"severity":"fatal","time":"2020-11-08T14:35:35Z"}

```
    "error": "config validation error:
        github.com/gitpod-io/gitpod/ws-proxy/cmd.getConfig
            /tmp/build/components-ws-proxy--app.ca3b7dcc9d8c3c1634ff4e7f1297c63007ce8d00/cmd/root.go:203
      - TheiaServer not configured:
        github.com/gitpod-io/gitpod/ws-proxy/pkg/proxy.(*TheiaServer).Validate
            /tmp/build/components-ws-proxy--app.ca3b7dcc9d8c3c1634ff4e7f1297c63007ce8d00/pkg/proxy/config.go:102",

```
```

**Comment by daimor** [comment]
registry-facade error
```
cannot source layer from eu.gcr.io/gitpod-io/self-hosted/supervisor:0.5.0: eu.gcr.io/gitpod-io/self-hosted/supervisor:0.5.0: not found
```

and proxy fails due to the issues with certificates
```
2020/11/08 14:40:09 [emerg] 1#1: BIO_new_file("/etc/nginx/certificates/fullchain.pem") failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/etc/nginx/certificates/fullchain.pem','r') error:2006D080:BIO routines:BIO_new_file:no such file)
nginx: [emerg] BIO_new_file("/etc/nginx/certificates/fullchain.pem") failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/etc/nginx/certificates/fullchain.pem','r') error:2006D080:BIO routines:BIO_new_file:no such file)
```
I'm not sure if it's my fault, and I've configured something wrong.

**Comment by daimor** [comment]
ws-proxy config validator expects Both theiaServer and blobServer to be present, but in chart it defined as only one of them

**Comment by daimor** [comment]
I'm manually updated ws-proxy-config, and added `theiaServer` there, so, my ws-proxy now running.
But I still have issues with `registry-facade` (cannot source layer from eu.gcr.io/gitpod-io/self-hosted/supervisor:0.5.0: eu.gcr.io/gitpod-io/self-hosted/supervisor:0.5.0: not found) and `proxy` due to missing certificates

**Comment by hm2075** [comment]
i think I may have came across this issue too

is this because of registry secrets?

do you need to create registry secrets?  are you using an external registry?

`kubectl create secret docker-registry image-builder-registry-secret --docker-server=xxx --docker-username=xxx --docker-password=xxx --docker-email=xxx`

and then use

image-builder-registry-secret  

in values.yaml


` registry:
      name: url_of_registry
      secretName: image-builder-registry-secret
`

and 

`
pullSecret:
      secretName: image-builder-registry-secret
`

**Comment by hm2075** [comment]
or

maybe 

kubectl create secret generic proxy-config-certificates --from-file=secrets/https-certificates    

where https-certificates contains PEM and CRT files

**Comment by daimor** [comment]
I don't use external registry, I supposed that aws-full-terraform, would install everything needed. Including SSL certificates for my domain I provided. But I don't see that it would be used. How it was supposed to work?

What I expected, is with some configuring, get own version of gitpod on my custom domain.

**Comment by z3ky** [comment]
> I'm manually updated ws-proxy-config, and added `theiaServer` there, so, my ws-proxy now running.
> But I still have issues with `registry-facade` (cannot source layer from eu.gcr.io/gitpod-io/self-hosted/supervisor:0.5.0: eu.gcr.io/gitpod-io/self-hosted/supervisor:0.5.0: not found) and `proxy` due to missing certificates

having the same problem, how did you update the ws-proxy-config?

**Comment by daimor** [comment]
Don't remember the exact steps. But something like this.

-  Get the actual YAML for configmap `kubectl get configmap ws-proxy-config -oyaml`
-  Placed to some file, edited it, cleaned from some staff added by kubernetes
-  And replaced old configmap with the new one. `kubectl replace -f ws-proxy-config.yml`

**Comment by z3ky** [comment]
Thank you, this helped me to get it running.
I added
`          "theiaServer": {
                "scheme": "http",
                "host": "theia-server.{{- .Release.Namespace -}}.svc.cluster.local:80",
                "staticVersionPathPrefix": ""
            },`
from the template and replaced {{- .Release.Namespace -}}

Did you solve the registry-facade error?

**Comment by daimor** [comment]
> Did you solve the registry-facade error?

No

**Comment by mamende** [comment]
I'm having the same proxy problems (as well as registry facade, which I was able to fix) as initially described with version 0.6.0 on OCI. Honestly, I don't know how to fix that with the descriptions in this thread. Could someone elaborate on this?

Will the two issues be fixed in upcoming releases?
